### PR TITLE
doc: implement hello_world_content_type example

### LINKED ISSUE #75: Implement example:  functions_http_content
Line numbers are approximate, they may change as the links are not pinned to a commit SHA:

https://github.com/GoogleCloudPlatform/golang-samples/blob/master/functions/http/http_content_type.go#L15-L74

https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/functions/helloworld/main.py#L18-L22

### PR DESCRIPTION
**Pull request by coryan**
Fixes #75

### CODE REVIEW

File: examples/site/hello_world_content/parse_www_form_urlencoded.cc [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+#include <algorithm>
+#include <charconv>
+#include <iterator>
+#include <map>
+#include <sstream>
+#include <string>
+#include <vector>
+
+std::string urldecode(std::string encoded);
```
Comment (devbww) [inline examples/site/hello_world_content/parse_www_form_urlencoded.cc:23]: Why not just define it here?

File: examples/site/hello_world_content/parse_www_form_urlencoded.cc [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  }();
+  std::map<std::string, std::string> result;
+  for (auto& tk : tokens) {
+    auto p = tk.find_first_of('=');
+    result.emplace(urldecode(tk.substr(0, p)), urldecode(tk.substr(p + 1)));
+  }
+  return result;
+}
+
+std::string urldecode(std::string encoded) {
```
Comment (devbww) [inline examples/site/hello_world_content/parse_www_form_urlencoded.cc:45]: Why pass by value?

File: examples/site/hello_world_content/hello_world_content.cc [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+#include <google/cloud/functions/http_request.h>
+#include <google/cloud/functions/http_response.h>
+#include <nlohmann/json.hpp>
+#include <map>
+#include <string>
+
+namespace gcf = ::google::cloud::functions;
+
+std::map<std::string, std::string> parse_www_form_urlencoded(
+    std::string const& text);
```
Comment (devbww) [inline examples/site/hello_world_content/hello_world_content.cc:25]: Can you have headers for separate translation units?

File: examples/site/hello_world_content/hello_world_content.cc [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+gcf::HttpResponse hello_world_content(gcf::HttpRequest request) {  // NOLINT
```
Comment (devbww) [inline examples/site/hello_world_content/hello_world_content.cc:27]: Didn't you deal with the `NOLINT` issue in another PR?

Comment (coryan) [reply examples/site/hello_world_content/hello_world_content.cc:34]: I started to, but it is not working yet.  I figured I could send this while I thought more about the other PR.

File: examples/site/hello_world_content/hello_world_content.cc (hunk 3 above)
Comment (coryan) [reply examples/site/hello_world_content/hello_world_content.cc:31]: Yes, but I just merged everything into one TU to simplify things.

File: examples/site/hello_world_content/parse_www_form_urlencoded.cc (hunk 2 above)
Comment (coryan) [reply examples/site/hello_world_content/parse_www_form_urlencoded.cc]: No reason, fixed.

File: examples/site/hello_world_content/parse_www_form_urlencoded.cc (hunk 1 above)
Comment (coryan) [reply examples/site/hello_world_content/parse_www_form_urlencoded.cc]: Fixed.

File: examples/site/hello_world_content/hello_world_content.cc [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+  }
+
+  gcf::HttpResponse response;
+  response.set_payload("Hello " + name);
+  response.set_header("content-type", "text/plain");
+  return response;
+}
+// [END functions_hello_world_content]
+
+namespace {
```
Comment (devbww) [inline examples/site/hello_world_content/hello_world_content.cc:57]: Can't we move these definitions up and avoid the forward declaration?

Comment (coryan) [reply examples/site/hello_world_content/hello_world_content.cc:57]: We could, but everything between the `// [START ...]` and `// [END ...]` markers goes straight to the example code:

https://cloud.google.com/functions/docs/writing/http#parsing_http_requests

I would prefer if that code is not cluttered with the implementation of `parse_www_form_urlencoded()`.

Comment (devbww) [reply examples/site/hello_world_content/hello_world_content.cc:57]: Ack.

**Review by devbww — APPROVED** [review]

**Pull request merged**